src/components/common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/StakingWidget/StakingWidgetProvider.tsx

```diff
@@ -0,0 +1,260 @@
+    throw new Error('Could not find StakingWidgetContext');
+  }
+
+  return ctx as StakingWidgetContextValues;
+};
+
+interface StakingWidgetProviderProps {
+  children: ReactNode;
+  motionData: MotionData;
+}
+
+const StakingWidgetProvider = ({
+  children,
+  motionData: {
+    motionId,
+    motionDomainId,
+    motionStakes: { nay: totalNAYStaked },
+    motionStakes,
+    rootHash,
+    skillRep,
+  },
+}: StakingWidgetProviderProps) => {
+  const { user } = useAppContext();
+  const { colony } = useColonyContext();
+  const [isSummary, setIsSummary] = useState(false);
+  const [isObjection, setIsObjection] = useState(false);
+
+  const { data: userReputation, loading: reputationLoading } =
+    useGetUserReputationQuery({
+      variables: {
+        input: {
+          colonyAddress: colony?.colonyAddress ?? '',
+          walletAddress: user?.walletAddress ?? '',
+          domainId: Number(motionDomainId),
+          rootHash,
+        },
+      },
+    });
+
+  const { minUserStake, requiredStake, loadingStakeData } =
+    useMinAndRequiredStakes(skillRep);
+
+  // TODO
+  //   const {
+  //     data: userData,
+  //     loading: userDataLoading,
+  //   } = useUserBalanceWithLockQuery({
+  //     variables: {
+  //       address: walletAddress,
+  //       tokenAddress: nativeTokenAddress,
+  //       colonyAddress,
+  //     },
+  //   });
+  //userData?.user?.userLock?.balance || 0,
+
+  const userActivatedTokens = useMemo(
+    () => new Decimal('10000000000000000000'),
+    [],
+  );
+
+  /* A user cannot stake more tokens than they have reputation. Thus, their rep is their max stake. */
+  const maxUserStake = useMemo(
+    () => new Decimal(userReputation?.getUserReputation ?? '0'),
+    [userReputation],
+  );
+
+  /* User has enough tokens to stake */
+  const enoughTokens = userActivatedTokens.gte(minUserStake);
+
+  /* User has enough reputation to stake */
+  const enoughReputation = maxUserStake.gt(0) && maxUserStake.gte(minUserStake);
+
+  const canUserStake = !!user && enoughReputation && enoughTokens;
+
+  const { yayPercentage, nayPercentage, totalPercentage } =
+    getStakedPercentages(isObjection, requiredStake, motionStakes);
+
+  const { remainingToStake, remainingToFullyNayStaked } = getRemainingToStake(
+    isObjection,
+    requiredStake,
+    motionStakes,
+  );
+
+  /*
+   * User's reputation in the domain is less than the amount still needed to stake, i.e. their
+   * staking ability is limited, not by their number of tokens activated, but by their (lack of) reputation.
+   */
+  const userNeedsMoreReputation =
+    userActivatedTokens.gt(maxUserStake) &&
+    maxUserStake.lt(remainingToStake.toString());
+
+  /*
+   * Occurs when the remaining amount to be staked is less than the user's minimum stake. (i.e. can't
+   * stake more than 100% of a motion)
+   */
+  const cantStakeMore =
+    remainingToStake.lte(minUserStake) && !remainingToStake.isZero();
+
+  const getErrorType = useCallback(
```

Comment:
These are the various states that will need testing

Reply:
These should also be some sort of type, possibly enums